### PR TITLE
CLDR-5854 Add-examples-for-relative-dates-and-times

### PR DESCRIPTION
**Pull request by haytenf**
CLDR-5854

| Locale               | Before               | After |
| -------------- | -------------- |  -------------- |
| cy |  Screenshot 2024-08-27 at 4 33 56 PM |  Screenshot 2024-08-27 at 4 34 32 PM |
| cy |  Screenshot 2024-08-27 at 4 35 20 PM | Screenshot 2024-08-27 at 4 50 53 PM |
| cy | Screenshot 2024-08-27 at 4 36 47 PM | Screenshot 2024-08-27 at 4 37 19 PM |
| cy | Screenshot 2024-08-27 at 4 43 45 PM | Screenshot 2024-08-27 at 4 46 19 PM |
| cy | Screenshot 2024-08-27 at 4 53 20 PM | Screenshot 2024-08-27 at 4 39 28 PM |
| ro | Screenshot 2024-08-27 at 4 56 03 PM |Screenshot 2024-08-27 at 4 57 18 PM |
| ro | Screenshot 2024-08-27 at 4 56 21 PM |Screenshot 2024-08-27 at 4 57 54 PM |
| ro | Screenshot 2024-08-27 at 5 02 07 PM | (future-few) Screenshot 2024-08-27 at 5 01 05 PM |
| ro | Screenshot 2024-08-27 at 4 59 38 PM  | Screenshot 2024-08-27 at 5 03 43 PM |
| it | Screenshot 2024-08-27 at 5 05 34 PM | Screenshot 2024-08-27 at 5 06 23 PM |
| it | Screenshot 2024-08-27 at 5 06 51 PM  | Screenshot 2024-08-27 at 5 07 40 PM |

- [x] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Comment by haytenf** [comment]
@macchiati @srl295 @btangmu Hello! I've made a new PR for this ticket since the old one was (accidentally) trying to merge to the wrong branch. The main thing I wanted to address is that the casing/context side isn't covered by this implementation for two main reasons (the examples for Romanian in the markdown table show this):
- the pattern I use for some of the examples is MMMMd, and when formatted using the rules for Romanian d comes first, so the capitalization becomes irrelevant
- simple date format produces the date appropriately lowercased for Romanian, which is then formatted with the relative item, so the middle-of-sentence capitalization issue doesn't arise

In terms of just getting examples in for the relative items, however, it may still be useful.

**Review by srl295 — APPROVED** [review]

File: tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java [hunk 1]
```diff
@@ -326,6 +326,19 @@ public void setCachingEnabled(boolean enabled) {
                 }
             };
 
+    // map relativeTimePattern counts to numeric examples
```
Comment (pedberg-icu) [inline tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java:329]: We might need to make this map dependent on locale. That can be done under [CLDR-18025](https://unicode-org.atlassian.net/browse/CLDR-18025) "Further work on examples for relative dates and times".

File: tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java [hunk 2]
```diff
@@ -3027,6 +3044,47 @@ void handleQuarters(XPathParts parts, String value, List<String> examples) {
...
+     * patterns as needed for relative dates vs relative times.
+     * Additionally, for relativeTimePattern items, ensure that
+     * numeric example corresponds to the count represented by the item.
+     */
+    private void handleRelative(
+            String xpath, XPathParts parts, String value, List<String> examples) {
+        String skeleton;
+        String type = parts.findAttributeValue("field", "type");
+        if (type.startsWith("hour")) {
+            skeleton = "Hm";
```
Comment (pedberg-icu) [inline tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java:3057]: We might want to make this "jm" or "Cm" to use hour cycle prefs based on locale. That can be done under [CLDR-18025](https://unicode-org.atlassian.net/browse/CLDR-18025) "Further work on examples for relative dates and times".

Comment (pedberg-icu) [reply tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java:3057]: Actually this is being used directly as a key to look up the availableFormats pattern, so mapping of meta characters like 'j' and 'C' will not work

File: tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java [hunk 3]
```diff
@@ -3027,6 +3044,47 @@ void handleQuarters(XPathParts parts, String value, List<String> examples) {
...
+        }
+        String checkPath =
+                "//ldml/dates/calendars/calendar[@type=\"gregorian\"]/dateTimeFormats/availableFormats/dateFormatItem[@id=\""
+                        + skeleton
+                        + "\"]";
+        String dateFormat = cldrFile.getWinningValue(checkPath);
+        SimpleDateFormat sdf = icuServiceBuilder.getDateFormat("gregorian", dateFormat);
+        String sampleDate = sdf.format(DATE_SAMPLE);
+        String example1 =
+                value.substring(0, 1).toUpperCase() + value.substring(1) + " (" + sampleDate + ")";
```
Comment (pedberg-icu) [inline tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java:3075]: Hmm. We need to use locale-sensitive Titlecasing (not just toUpperCase the first char), and we also only want to do this if the contextTransforms data says that relative fields should be title-cased in a stand-alone context. Again, I will fix this in [CLDR-18025](https://unicode-org.atlassian.net/browse/CLDR-18025) "Further work on examples for relative dates and times".

**Pull request merged**